### PR TITLE
topology: add guide for specify DSP cores

### PR DESCRIPTION
**Pull request by keyonjie**
Add guide and example to explain how to specify DSP core index for
pipeline or widgets/components.

Signed-off-by: Keyon Jie <yang.jie@linux.intel.com>

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]

File: developer_guides/topology/topology.rst [hunk 1]
```diff
@@ -292,6 +292,45 @@ where:
   can be chosen predefined configurations such as MONO_PDM0_MICA,
   STEREO_PDM0, FOUR_CH_PDM0_PDM1 etc.
 
+.. _dsp-core-in-topology:
+
+1.7 DSP Core Index
+------------------
+
+We can specify a DSP Core Index that a pipeline or component will be scheduled
+on in the topology file.
```
Comment (plbossart) [inline developer_guides/topology/topology.rst:301]: The topology file can specify on which DSP core a pipeline or component will be scheduled on.

File: developer_guides/topology/topology.rst [hunk 2]
```diff
@@ -612,4 +651,4 @@ actually push both messages to a dot file:
 .. _here: https://www.alsa-project.org/main/index.php/ALSA_topology
 .. _SOFT: https://github.com/thesofproject/soft
 
-.. comment "master" has been replaced with "main"
\ No newline at end of file
+.. comment "master" has been replaced with "main"
```
Comment (plbossart) [inline developer_guides/topology/topology.rst:654]: what is this modification about?

File: platforms/intel-cavs/commons/multicore-processing.rst [hunk 3]
```diff
@@ -58,4 +58,6 @@ stop**.
 
 .. note:: The kernel also needs to enable cores on the host side, before
           even sending the ``SOF_IPC_PM_CORE_ENABLE`` IPC to the FW.
+          For how to specify core usage in topology file, refer to the
```
Comment (plbossart) [inline platforms/intel-cavs/commons/multicore-processing.rst:61]: for details on to specify....

**Review by plbossart — CHANGES_REQUESTED** [review]
wording improvements suggested below

**Comment by lgirdwood** [comment]
@plbossart fwiw, @deb-intel always re-flows PR's for grammar after merge and before publishing. We couldn't do without her :)

**Comment by plbossart** [comment]
> @plbossart fwiw, @deb-intel always re-flows PR's for grammar after merge and before publishing. We couldn't do without her :)

I know but I felt the wording was ambiguous even for our talented technical writers.

File: developer_guides/topology/topology.rst (hunk 2 above)
Comment (keyonjie) [reply developer_guides/topology/topology.rst:654]: don't know why this happen, looks it is introduced automatically when I edit this .rst file and not removable.

**Review by plbossart — APPROVED** [review]

**Review by deb-intel — APPROVED** [review]
Will make grammatical edits in a separate PR.

File: developer_guides/topology/topology.rst (hunk 2 above)
Comment (deb-intel) [reply developer_guides/topology/topology.rst:651]: @plbossart and @keyonjie That was a note I made to indicate that I had replaced references to "master" with "main." It was to make sure I used the correct change. I can remove it.

**Pull request merged**